src/utils/wallet.js

```diff
@@ -438,53 +647,266 @@ class Wallet {
+            ...fak2lak.map((pk) => nearApiJs.transactions.addKey(pk, nearApiJs.transactions.functionCallAccessKey(accountId, METHOD_NAMES_LAK, null))),
+            // confirmOnlyKey
+            nearApiJs.transactions.addKey(confirmOnlyKey, nearApiJs.transactions.functionCallAccessKey(accountId, METHOD_NAMES_CONFIRM, null)),
+            // deploy and initialize contract
+            nearApiJs.transactions.deployContract(contractBytes),
+            nearApiJs.transactions.functionCall('new', newArgs, 10000000000000, '0'),
+        ]
+        console.log('actions', actions)
+        return await account.signAndSendTransaction(accountId, actions);
+    }
 
+    async createNewAccountForTempAccount(accountId, fundingContract, fundingKey) {
+        // create account because recovery is validated
+        if (!accountId) {
+            // updated to get tempAccount.accountId
+            accountId = this.getAccountId()
+        }
+        await this.createNewAccount(accountId, fundingContract, fundingKey)
+        const account = this.getAccount(accountId)
+        console.log('account created', account)
+        // remove the temp account, we now have a real account on chain
+        delTempAccount()
+        return account
+    }
+
+    async setupRecoveryMessage(accountId, method, securityCode, fundingContract, fundingKey) {
+        // temp account was set during create account
+        console.log('setupRecoveryMessage', accountId)
+        let securityCodeResult = await this.validateSecurityCode(accountId, method, securityCode);
+        if (!securityCodeResult || securityCodeResult.length === 0) {
+            console.log('INVALID CODE', securityCodeResult)
+            return
+        }
+        const tempAccount = getTempAccount()
+        // if this is a tempAccount we'll create a new account now
+        if (tempAccount && tempAccount.accountId) {
+            // added method above, also used for seed recovery
+            await this.createNewAccountForTempAccount(accountId, fundingContract, fundingKey)
+        }
+        // now send recovery seed phrase
+        const { seedPhrase, publicKey } = generateSeedPhrase();
+        const { account, has2fa } = await this.getAccountAndState(accountId)
+        const accountKeys = await account.getAccessKeys();
+        if (has2fa) {
+            await this.addAccessKey(account.accountId, account.accountId, splitPK(publicKey))
+            // const request = {
+            //     receiver_id: account.accountId,
+            //     actions: [{ type: 'AddKey', public_key: splitPK(publicKey) }]
+            // }
+            // await this.makeTwoFactorRequest(account, request)
+        } else {
+            if (!accountKeys.some(it => it.public_key.endsWith(publicKey))) {
+                await account.addKey(publicKey);
+            }
+        }
+        /********************************
+        Can we send signed JSON here? Should we?
```

Comment:
there is already a check on near-contract-helper side that seed phrase matches the key added to the account

Reply:
removed thanks